dulwich/refs.py

```diff
@@ -713,19 +713,21 @@ def set_if_equals(self, name, old_ref, new_ref, committer=None,
 
         ensure_dir_exists(os.path.dirname(filename))
         with GitFile(filename, 'wb') as f:
+            try:
+                # read again while holding the lock
+                orig_ref = self.read_loose_ref(realname)
+                if orig_ref is None:
```

Comment:
There is no lock on the packed refs, how does this ensure that packed refs isn't changing at the same time? Since it will no longer write the new ref file.